odoo/addons/base/tests/test_ir_actions.py

```diff
@@ -383,3 +383,63 @@ def test_unlink_with_dependant(self):
         # uninstall mode: unlink dependant fields
         field.with_context(_force_unlink=True).unlink()
         self.assertFalse(dependant.exists())
+
+    def test_related_field(self):
+        """ create a related field, check filled values"""
+        SOURCE_MODEL = 'res.users'
+        SOURCE_FIELD = 'login'
+
+        TARGET_MODEL = 'res.partner'
```

Comment:
You are playing with very specific models, and IMHO using `res.users` is not a good idea.
It got me all confused at first because `res.users` inherits from `res.partner`.
Also your terminology of "source" and "target" models was also confusing to me.

I suggest to use `res.partner` and `res.country`, and call them the "model" and "comodel",
since they are the model and comodel of the join field `country_id`, and add a related field "country name".
Moreover, every instance of Odoo has a bunch of records in `res.country`, so no need to create them.

Reply:
I was confused on writing the tests too :) Then I added some comments and decided that it's clear enough.